feat(ocpp21): Add update allowed energy transfer modes callback (#1096)
EVerest/libocpp: doc/v2/ocpp_2x_status.md

| ID        | Status                                          | Remark |
| --------- | ----------------------------------------------- | ------ |
|           | AllowedEnergyTransfer / RequestedEnergyTransfer |        |
| Q01.FR.01 |   ✅ / 🌐                                       |        |
| Q01.FR.02 |   ⛽️                                            | This is handled in OCPP2 module in EVerest. |
| Q01.FR.03 |   ✅ / ⛽️                                       | Information for the message is filled in the OCPP2 module and libocpp sends the message. |
| Q01.FR.04 |   🌐                                            |        |

> | Q01.FR.05 |   ❎                                            | In principle it is a charging station requirement, but should never occur / is not really possible to renegotiate at this time.       |

Reviewer: Shouldnt we keep this open since its just not supported by libocpp at the moment?
```suggestion
| Q01.FR.05 |                                              | In principle it is a charging station requirement, but should never occur / is not really possible to renegotiate at this time.       |
```

| Q01.FR.06 |   ✅                                            |        |
| Q01.FR.07 |   🌐                                            |        |
| Q01.FR.08 |   🌐                                            |        |

> | Q01.FR.09 |   ⛽️                                            | Information for the message is filled in the OCPP2 module and libocpp sends the message. |

Reviewer: same like for Q01.FR.03
```suggestion
| Q01.FR.09 |   ✅                                           | The consumer of libocpp shall call `on_ev_charging_needs` when ChargeParameterDiscoveryReq is received via ISO15118 |
```

|           | Device model                                    | User should fill DM. |
| Q01.FR.30 |   ⛽️                                            | User should fill DM. |
| Q01.FR.31 |   ⛽️                                            | User should fill DM. |
| Q01.FR.32 |   ⛽️                                            | User should fill DM. |
| Q01.FR.33 |                                                 |        |
| Q01.FR.34 |                                                 |        |
| Q01.FR.35 |                                                 |        |
| Q01.FR.36 |   ⛽️                                            | User should fill DM. |

## Bidirectional Power Transfer - Charging only (V2X control) before starting V2X (New in OCPP 2.1)

| ID        | Status | Remark |
| --------- | ------ | ------ |
| Q02.FR.01 |   🌐   |        |
| Q02.FR.02 |   🌐   |        |
| Q02.FR.03 |   🌐   |        |

> | Q02.FR.04 | ✅ / ⛽️|        |

Reviewer: ```suggestion
| Q02.FR.04 |   ✅                                           | The consumer of libocpp shall call `on_ev_charging_needs` when ChargeParameterDiscoveryReq is received via ISO15118 |
```

| Q02.FR.05 |   🌐   |        |
| Q02.FR.06 |   ✅   |        |

> | Q02.FR.07 | ✅ / ⛽️|        |

Reviewer: ```suggestion
| Q02.FR.07 |   ✅                                           | The consumer of libocpp shall call `on_ev_charging_needs` when ChargeParameterDiscoveryReq is received via ISO15118 |
```
